sys-apps/firejail/firejail-0.9.62.4.ebuild

```diff
@@ -0,0 +1,85 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs
+
+if [[ ${PV} != 9999 ]]; then
+	KEYWORDS="~amd64 ~x86"
+	SRC_URI="https://github.com/netblue30/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+else
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/netblue30/firejail.git"
+	EGIT_BRANCH="master"
+fi
+
+DESCRIPTION="Security sandbox for any type of processes"
+HOMEPAGE="https://firejail.wordpress.com/"
+
+LICENSE="GPL-2"
+SLOT="0"
+IUSE="apparmor +chroot contrib debug +file-transfer +globalcfg +network +overlayfs +private-home +seccomp +suid test +userns vim-syntax +whitelist x11"
+RESTRICT="!test? ( test )"
+
+RDEPEND="apparmor? ( sys-libs/libapparmor )"
+
+DEPEND="${RDEPEND}
+	!sys-apps/firejail-lts
+	test? ( dev-tcltk/expect )"
+
+PATCHES=(
+	"${FILESDIR}"/"${P}"-nvidia-no-nogroups.patch
+	"${FILESDIR}"/"${P}"-apparmor-global.patch
+)
+
+src_prepare() {
+	default
+
+	find ./contrib -type f -name '*.py' | xargs sed --in-place 's-#!/usr/bin/python3-#!/usr/bin/env python3-g' || die
```

Comment:
Instead of the ebuild trying to handle Python shebangs and the build system installing them can these be installed with a `python_*script` function from `python-utils-r1.eclass`?

https://devmanual.gentoo.org/eclass-reference/python-utils-r1.eclass/index.html

Reply:
Huh, actually, I could use help figuring out the right thing to do here. The `find | xargs sed` is not the right thing, but...

The only python scripts in firejail are in `contrib/`, and only installed if `USE=contrib` (so perhaps any mangling should be `if use contrib`'ed). All but one of those scripts already use `#!/usr/bin/env python3` as their shebang. Just because they should probably all be similar, I issued a PR upstream to change that one to be like all the others (https://github.com/netblue30/firejail/pull/3672).

But... that seems to be the opposite of what `python-utils.r1.eclass` functions want: `python_doscript`, `python_newscript`, `python_fix_shebang` all say "Only files having a 'python*' shebang will be modified", so they probably would not act on `#!/usr/bin/env python3`?

So should I in fact invert the current `find | xargs`, have it convert *from* env *to* a python shebang, and then call a `python_*script` function?

Reply:
OK, @ajakk I've updated the PR to address everything except this bit. Thanks for the review, please lmk about this part.

Reply:
It looks to me like you can drop that `find` command in src_prepare, disable the `contrib` configure options so the build system doesn't install it, then install those conditionally with `python_doscript`.

Reply:
OK, did mostly that. I found and worked around a few issues:

- ended up using `python-single-r1` so that EPYTHON would be set for me
- `python_doscript` would not install without conforming shebangs
- `python_fix_shebang` would not fix upstream's shebangs without -f
- `python_fix_shebang` choked on one contrib/*.py that was all `^M`'ed, so had to clean that up first
- the contrib/ scripts aren't safe to install in `/usr/bin/` - generic names that could collide. So I used `python_scriptinto` to install to the path that the `contrib-install` configure option used (plus, now they will be where upstream expects them for support purposes).

So, provided the way I handled those are acceptable, everything should now be resolved I think.